src/clients/c++/request_http.cc

```diff
@@ -632,28 +650,40 @@ SharedMemoryControlHttpContextImpl::RegisterSharedMemory(
     const std::string& name, const std::string& shm_key, const size_t offset,
     const size_t byte_size)
 {
-  return SendRequest("register", name, shm_key, offset, byte_size);
+  return SendRequest("register", name, shm_key, offset, byte_size, nullptr);
 }
 
 Error
 SharedMemoryControlHttpContextImpl::UnregisterSharedMemory(
     const std::string& name)
 {
-  return SendRequest("unregister", name, "", 0, 0);
+  return SendRequest("unregister", name, "", 0, 0, nullptr);
 }
 
 Error
 SharedMemoryControlHttpContextImpl::UnregisterAllSharedMemory()
 {
-  return SendRequest("unregisterall", "", "", 0, 0);
+  return SendRequest("unregisterall", "", "", 0, 0, nullptr);
+}
+
+Error
+SharedMemoryControlHttpContextImpl::GetSharedMemoryStatus(
+    SharedMemoryStatus* shm_status)
+{
+  return SendRequest("status", "", "", 0, 0, shm_status);
 }
 
 Error
 SharedMemoryControlHttpContextImpl::SendRequest(
     const std::string& action_str, const std::string& name,
-    const std::string& shm_key, const size_t offset, const size_t byte_size)
+    const std::string& shm_key, const size_t offset, const size_t byte_size,
+    SharedMemoryStatus* shm_status)
```

Comment:
Instead of passing shm_status into this common function, just move the response_ -> status into the calling function (GetSharedMemoryStatus)

Reply:
Done